components/selectDevice/Device.tsx

```diff
@@ -0,0 +1,53 @@
+"use client";
+
+import { useRouter } from "next/navigation";
+import { useState, useTransition } from "react";
+
+import transferPlayback from "@/lib/device/transferPlayback";
+
+const Device = ({
+  device,
+  currentDeviceId,
+  isPlaying,
+}: {
+  device: SpotifyApi.UserDevice;
+  currentDeviceId: string;
+  isPlaying: boolean;
```

Comment:
It says in the docs "false or not provided: keep the current playback state." so you don't need this either (which lets you get rid of getPlaybackState)